code/core/arm9/source/GbaColorCorrection/ColorLut.cpp

```diff
@@ -0,0 +1,116 @@
+#include "common.h"
+#include "ColorLut.h"
+#include "GammaLut.h"
+
+// Based on https://gist.github.com/profi200/bfa7be60b3eecb8c43f59000f626c743
+
+u16 gColorLut[COLOR_LUT_SIZE] __attribute__((section(".lutram")));
+
+const ColorProfile* gCurrentPreset = &Agb001; // Default is AGB_001
+
+// When the 2d engine converts from 5 to 6 bit, the lsb bit will always be zero (i.e. 31 -> 62)
+inline constexpr u32 rgb8ToRgb5(u32 value8)
+{
+    u32 value5 = (value8 * 63 + 255) / (255 * 2);
+    if (value5 > 31)
+        return 31;
+    return value5;
+}
+
+// Convert RGB8 to RGB6 (for the 6-bit green)
+inline constexpr u32 rgb8ToRgb6(u32 value8)
+{
+    return (value8 * 63 + 128) / 255;
+}
+
+// Simple and optimal clamping
+inline constexpr u8 clamp255(int val) 
+{
+    return val < 0 ? 0 : (val > 255 ? 255 : val);
+}
+
+// Apply correction matrix from selected color profile
+inline void applyColorMatrix(const fix32<12> matrix[3][3], fix32<12> r, fix32<12> g, fix32<12> b, fix32<12>& outR, fix32<12>& outG, fix32<12>& outB)
```

Comment:
```suggestion
static void applyColorMatrix(const fix32<12> matrix[3][3], fix32<12> r, fix32<12> g, fix32<12> b, fix32<12>& outR, fix32<12>& outG, fix32<12>& outB)
```
The compiler can figure out inlining just fine for statics